### PR TITLE
doc(nyz): add impala benchmark and polish doc

### PR DESCRIPTION
**Pull request by PaParaZz1**

### CODE REVIEW

File: source/hands_on/impala.rst [hunk 1]
```diff
@@ -311,18 +338,41 @@ process is as follows.
...
+     - 20
+     - .. image:: images/benchmark/impala_pong.png
+     - `config_link_p <https://github.com/opendilab/DI-engine/blob/main/dizoo/atari/config/serial/pong/pong_impala_config.py>`_
+     - | IMPALA paper shallow 200M (20.4)
+   * - | Qbert
+       | (QbertNoFrameskip-v4)
+     - 13175
+     - .. image:: images/benchmark/impala_qbert.png
+     - `config_link_q <https://github.com/opendilab/DI-engine/blob/main/dizoo/atari/config/serial/qbert/qbert_impala_config.py>`_
+     - | MPALA paper shallow 200M (18901)
```
Comment (hiha3456) [inline source/hands_on/impala.rst:364]: Just IMPALA here xD

File: source/hands_on/impala.rst [hunk 2]
```diff
@@ -311,18 +338,41 @@ process is as follows.
...
+   * - | SpaceInvaders
+       | (SpaceInvadersNoFrame skip-v4)
+     - 977
+     - .. image:: images/benchmark/impala_spaceinvaders.png
+     - `config_link_s <https://github.com/opendilab/DI-engine/blob/main/dizoo/atari/config/serial/spaceinvaders/spaceinvaders_impala_config.py>`_
+     - | MPALA paper shallow 200M (1726)
```
Comment (hiha3456) [inline source/hands_on/impala.rst:370]: Just IMPALA here xD

**Review by hiha3456 — COMMENTED** [review]
Excellent doc which is easy to understand and covers most of the main points for IMPALA. The algorithm is difficult for me, but the explanation here is very detailed and easy to understand. I just left some comments for the little issues of spelling😉.

File: source/hands_on/impala.rst [hunk 3]
```diff
@@ -180,18 +205,20 @@ refer to doc of ``ding.rl_utils.adder``.
...
     with length smaller than ``unroll_len``. This method is kind of naive and usually is not a good choice. Since in
     Reinforcement Learning, the last few data in an episode is usually very important, we can't just throw away them.
 
-    The second method is ``last``, which means if the total length trajectory is smaller than ``unrollen_len``,
+    2. The second method is ``last``, which means if the total length trajectory is smaller than ``unroll_len``,
     we will use zero padding. Else, we will use data from previous piece to pad residual piece. This method is set as
     default and recommended.
 
-    The last method ``null_padding`` is just zero padding, which is not vert efficient since many data will be ``null``.
+    3. The last method ``null_padding`` is just zero padding, which is not vert efficient since many data will be ``null``.
```
Comment (Will-Nie) [inline source/hands_on/impala.rst:216]: 3 is not aligned well with 1, and 2

File: source/hands_on/impala.rst [hunk 4]
```diff
@@ -47,53 +51,59 @@ at time s is defined as follows:
...
 .. math::
 
-    Loss_value &= (v_s - V_\theta(x_s))^2 \\
-    Loss_policy &= -\rho_s \log \pi_\phi(a_s \vert x_s)  \big(r_s + \gamma v_{s+1} - V_\theta(x_s)\big) \\
-    Loss_entropy &= -H(\pi_\phi) = \sum_a \pi_\phi(a\vert x_s)\log \pi_\phi(a\vert x_s)
+    loss_{value} &= (v_s - V_\theta(x_s))^2 \\
+    loss_{policy} &= -\rho_s \log \pi_\phi(a_s \vert x_s)  \big(r_s + \gamma v_{s+1} - V_\theta(x_s)\big) \\
+    loss_{entropy} &= -H(\pi_\phi) = \sum_a \pi_\phi(a\vert x_s)\log \pi_\phi(a\vert x_s)
 
 where :math:`H(\pi_{\phi})`, entropy of policy :math:`\phi`, is an bonus to encourage exploration.
```
[the remark this answers is not in the document]
Comment (chenyingvivian) [reply source/hands_on/impala.rst]: 没有修改

File: source/hands_on/impala.rst [hunk 5]
```diff
@@ -4,39 +4,43 @@ IMPALA
 Overview
 ---------
 IMPALA, or the Importance Weighted Actor Learner Architecture, is an off-policy actor-critic framework that
-decouples acting from learning and learns from experience trajectories using V-trace. This method is first
+decouples data collecting from learning and optimizes policy from experience trajectories using off-policy correction V-trace. This method is first
```
[the remark this answers is not in the document]
Comment (PaParaZz1) [reply source/hands_on/impala.rst:7]: fixed it

File: source/hands_on/impala.rst [hunk 6]
```diff
@@ -4,39 +4,43 @@ IMPALA
...
-1. Impala  is a **model-free** and **off-policy** RL algorithm.
+1. Impala is a **model-free** and **off-policy** RL algorithm.
 
 2. Impala can support both **discrete** action spaces and **continuous** action spaces.
 
-3. Impala is a actor-critic RL algorithm, which optimizes actor network and critic network, respectively.
+3. Impala is a actor-critic RL algorithm with value network, which optimizes actor network and critic (value) network, respectively.
 
-4. Impala decouples acting from learning. Collectors in impala will not compute value or advantage.
+4. Impala can take advantages of the old off-policy data with corresponding **off-policy correction**  to stabilize learning.
```
[the remark this answers is not in the document]
Comment (PaParaZz1) [reply source/hands_on/impala.rst:19]: fixed it

File: source/hands_on/impala.rst [hunk 7]
```diff
@@ -4,39 +4,43 @@ IMPALA
...
 
+5. Impala decouples data collecting from learning. Collectors in impala will not compute value or advantage.
```
[the remark this answers is not in the document]
Comment (PaParaZz1) [reply source/hands_on/impala.rst:21]: fixed it

**Pull request merged**